scipy/stats/_stats_py.py

```diff
@@ -98,7 +98,7 @@ def _contains_nan(a, nan_policy='propagate'):
         # This can happen when attempting to sum things which are not
         # numbers (e.g. as in the function `mode`). Try an alternative method:
         try:
-            contains_nan = np.nan in set(a.ravel())
+            contains_nan = np.any(np.isnan(a.ravel()))
```

Comment:
Is `ravel` actually needed?

Reply:
I believe so, assuming this function is ever used by non-1d arrays. If that is not the case anywhere in the package it could likely be removed.

Reply:
It is not needed. `np.isnan` returns `True` wherever an element within the array is `nan`. `np.any` returns `True` if any element in the array is `True`.

Reply:
A quick test also revealed this, you are in fact sir, correct. I wasn't sure if it was for speed issues, but because it was in the original code I retained it, feel free to remove.